src/backend/gsroa.c

```diff
@@ -220,9 +299,9 @@ void sliceStructs()
                     sold->Spreg2 = NOREG;
                 }
                 type_free(sold->Stype);
-                sold->Stype = type_fake(TYnptr);
+                sold->Stype = type_fake(sia[si].ty0);
                 sold->Stype->Tcount++;
-                snew->Stype = type_fake(TYnptr);
+                snew->Stype = type_fake(sia[si].ty1);
```

Comment:
Because they may be different types. a.ptr and a.length are different, for example.